docs/cadence/language/built-in-functions.mdx

```diff
@@ -29,39 +29,43 @@ Use this function for internal sanity checks.
 
 The message argument is optional.
 
-## `unsafeRandom`
+## `revertibleRandom`
 
 ```cadence
-fun unsafeRandom(): UInt64
+fun revertibleRandom(): UInt64
 ```
 
+<Callout type="info">
+`unsafeRandom()` function currently behaves the same way as `revertibleRandom()`.
+`unsafeRandom` will be deprecated in the next release of Cadence. 
+</Callout>
+
 Returns a pseudo-random number.
 
-<Callout type="warning">
-Smart contract developers should be mindful about the limitations of unsafeRandom.
-The stream of random numbers produced is potentially unsafe in the following two regards:
+The random number returned is safe as it is backed by a distributed source of randomness,
+that is unbiased, unpredictable and verifiable.
 
-1. The sequence of random numbers is potentially predictable by transactions within the same block
-and by other smart contracts calling into your smart contract.
-2. A transaction calling into your smart contract can potentially bias the sequence of random numbers which
-your smart contract internally generates.
+The random number cannot be biased or predicted by miners or by other transactions or code
+running before a call to `revertibleRandom`.
 
-The Flow project is working towards removing these limitations incrementally.
-Once Flow addressed these points and randomness is safe,
-the "unsafe" qualifier is going to get removed.
+Moreover, the sequence of returned random numbers is different per block, and per transaction
+within the same block.
```

Comment:
I added your suggestion with what I think is minor rewording. 

I see the advantage of linking the forum post for details, but I would have preferred if official docs could link to a more official post in the site rather than a forum message. My preference is not to add the link but it's not a strong preference.

Reply:
We can always update the link to the proper blog post once it is out. So far, I think it is better to have some reference than none.

Reply:
Added the link as per our offline discussion